src/T2DMap.cpp

```diff
@@ -672,13 +672,503 @@ bool T2DMap::sizeFontToFitTextInRect( QFont & font, const QRectF & boundaryRect,
+    }
+
+    painter.restore();
+    if (!pArea->gridMode) {
+        QMapIterator<int, QPoint> it(mAreaExitsList);
+        while (it.hasNext()) {
+            it.next();
+            QPoint P = it.value();
+            int rx = P.x();
+            int ry = P.y();
+
+            QRectF dr = QRectF(rx, ry, mRoomWidth * rSize, mRoomHeight * rSize);
+
+            // clang-format off
+            if (((mPick || picked)
+                 && mPHighlight.x() >= (dr.x() - mRoomWidth / 3.0)
+                 && mPHighlight.x() <= (dr.x() + mRoomWidth / 3.0)
+                 && mPHighlight.y() >= (dr.y() - mRoomHeight / 3.0)
+                 && mPHighlight.y() <= (dr.y() + mRoomHeight / 3.0))
+                && mStartSpeedWalk) {
+
+                // clang-format on
+                mStartSpeedWalk = false;
+                // This draws a red circle around the out of area exit that
+                // was choosen as the target for the speedwalk, but it is
+                // only shown for one paintEvent call and it is not obvious
+                // that it is useful, note that there is similar code for a
+                // room being clicked on that is WITHIN the area, that is
+                // above this point in the source code:
+                float roomRadius = (0.8 * mRoomWidth) / 2.0;
+                QPointF roomCenter = QPointF(rx, ry);
+                QRadialGradient gradient(roomCenter, roomRadius);
+                gradient.setColorAt(0.95, QColor(255, 0, 0, 150));
+                gradient.setColorAt(0.80, QColor(150, 100, 100, 150));
+                gradient.setColorAt(0.799, QColor(150, 100, 100, 100));
+                gradient.setColorAt(0.7, QColor(255, 0, 0, 200));
+                gradient.setColorAt(0, Qt::white);
+                QPen transparentPen(Qt::transparent);
+                QPainterPath myPath;
+                painter.setBrush(gradient);
+                painter.setPen(transparentPen);
+                myPath.addEllipse(roomCenter, roomRadius, roomRadius);
+                painter.drawPath(myPath);
+
+                mPick = false;
+                mTarget = it.key();
+                if (mpMap->mpRoomDB->getRoom(mTarget)) {
+                    mpMap->mTargetID = mTarget;
+                    if (mpMap->findPath(playerRoomId, mpMap->mTargetID)) {
+                        mpMap->mpHost->startSpeedWalk();
+                    } else {
+                        QString msg = tr("Mapper: Cannot find a path to this room using known exits.\n");
+                        mpHost->mpConsole->printSystemMessage(msg);
+                    }
+                }
+            }
+        }
+    }
+}
+
 // Revised to use a QCache to hold QPixmap * to generated images for room symbols
 void T2DMap::paintEvent(QPaintEvent* e)
 {
     if (!mpMap) {
         return;
     }
-    bool __Pick = mPick;
+    // As it happens this local is never changed in this method so it can be
+    // made const:
+    const bool __Pick = mPick;
```

Comment:
```suggestion
    const bool pick = mPick;
```

Reply:
Agreed! Whoever thought to using a double leading underscore is heading for trouble as far as I can see, e.g. https://stackoverflow.com/questions/224397/why-do-people-use-double-underscore-so-much-in-c - mind you, I think I have used them in the qmake project file but then that is not text that is parsed by the compiler/linker...